src/expand.cpp

```diff
@@ -98,6 +100,8 @@ namespace Sass {
       return k;
     }
 
+    this->at_root_without_rule = false;
```

Comment:
On a third look I don't see why it shouldn't work here if you add the `LOCAL_FLAG` call here ... Since you seem to reset it just right before you exit the function (and therefore this variable scope).